File: tks/RepositoriesAdapters/src/main/kotlin/mappers/RentMapper.java
package mappers;

import data.RentEntity;
import model.Rent;
import org.mapstruct.MappingTarget;
import org.mapstruct.factory.Mappers;

public interface RentMapper {

    RentMapper INSTANCE = Mappers.getMapper(RentMapper.class);

    Rent mapEntityToDomainObject(RentEntity entity);

    RentEntity mapDomainObjectToEntity(Rent user);

    void mapDomainObjectToEntity(Rent user, @MappingTarget RentEntity entity);
}

File: tks/RepositoriesAdapters/src/main/kotlin/mappers/RentMapper.kt
package mappers

import common.mappers.EmailMapper
import data.RentEntity
import model.Rent
import org.mapstruct.MappingTarget
import org.mapstruct.Mapper
import org.mapstruct.factory.Mappers

@Mapper(uses = [EmailMapper::class])
interface RentMapper {
    fun mapEntityToDomainObject(entity: RentEntity?): Rent?
    fun mapDomainObjectToEntity(user: Rent?): RentEntity?
    fun mapDomainObjectToEntity(user: Rent?, @MappingTarget entity: RentEntity?)

    companion object {
        val INSTANCE = Mappers.getMapper(RentMapper::class.java)
    }
}